src/Fmt_ast.ml

```diff
@@ -1161,10 +1161,19 @@ and fmt_pattern c ?pro ?parens ({ctx= ctx0; ast= pat} as xpat) =
            (fmt "exception@ " $ fmt_pattern c (sub_pat ~ctx pat)))
   | Ppat_extension ext -> hvbox 2 (fmt_extension c ctx "%" ext)
   | Ppat_open (lid, pat) ->
+      let opn, cls =
+        let can_skip_parens =
+          match pat.ppat_desc with
+          | Ppat_array _ | Ppat_record _ -> true
+          | _ -> (
```

Comment:
Shouldn't there be a case for tuples, which consults parens_tuple_patterns? Consider cases such as:
```
[@@@ocamlformat "parens-tuple-patterns=always"]

let M.((x, y)) = ()
```

Reply:
Indeed it should